src/RemoteInboxNotifications/EvaluateAndGetStatus.php

```diff
@@ -29,7 +29,9 @@ public static function evaluate( $spec, $current_status, $stored_state, $rule_ev
 			return $current_status;
 		}
 
-		$evaluated_result = $rule_evaluator->evaluate( $spec->rules, $stored_state );
+		$evaluation_logger = new EvaluationLogger( $spec->slug );
+		$evaluated_result  = $rule_evaluator->evaluate( $spec->rules, $stored_state, $evaluation_logger );
```

Comment:
The `RuleEvaluator` doesn't have access to the spec slug, unfortunately. That's the only reason I'm initializing the class here and passing the class as an argument.

Reply:
What do you think about creating an optional `$logger_args` argument instead of passing the whole `EvaluationLogger`?  This way, we can pass a `slug` or any `key` and also the `source` used so it doesn't always default to `remote-inbox-notifications`.

This has a couple of advantages:
* Consumers don't need to handle logic around the evaluator itself
* Key/slugs could be optional
* We can update other items like the `source` to use this everywhere rules are evaluated

Reply:
@joshuatf That's a great suggestion. I'll make a change.